### PR TITLE
[WIP] Adding new linter rule 'import/no-unused-modules' & amending code as needed to resolve issues

### PR DESCRIPTION
**Pull request by Shazib**
- I think import doesn't work by default with typescript so i had to add an extra package for typescript support
- I've disabled checking on the `/icons` subfolder because it warns about every unused icon
- I've temporarily disabled it on `loot-core` because it was returning a lot of false positives. (help?)
- I have not compared benchmarked performance but it doesn't feel too slow.

Code Changes:
- Changes to mobile pages were a separate commit as i know this is still WIP so we can undo these
- Ignored the plaid file as i know this is also still wip.
- Hopefully the rest all makes sense.

### CODE REVIEW

**Comment by biohzrddd** [comment]
Very interested in getting this integrated! It will save time updating stale components.

**Comment by j-f1** [comment]
> I think import doesn't work by default with typescript so i had to add an extra package for typescript support

Good catch!

> I've temporarily disabled it on loot-core because it was returning a lot of false positives. (help?)


There are >250 places in `desktop-client` where we import a thing directly from `loot-core/src/*`. Not sure what the best way forward is to deal with this but it would be nice to clean up unused code here too.

File: packages/desktop-client/src/plaid.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint import/no-unused-modules: 0 */
```
[the remark this answers is not in the document]
Comment (Shazib) [reply packages/desktop-client/src/plaid.js]: I think this is the only place i'm doing this

**Comment by Shazib** [comment]
> > I think import doesn't work by default with typescript so i had to add an extra package for typescript support
> 
> Good catch!
> 
> > I've temporarily disabled it on loot-core because it was returning a lot of false positives. (help?)
> 
> There are >250 places in `desktop-client` where we import a thing directly from `loot-core/src/*`. Not sure what the best way forward is to deal with this but it would be nice to clean up unused code here too.

I'm not sure what exactly the problem is tbh, maybe some problem with not being able to detect .`.ts` inports used from `.js`? Or the fact that they're in different projects? But `loot-core` reports around 300 errors for things that _are_ used in `desktop-client`

**Comment by Shazib** [comment]
Putting this back to WIP as i have more code changes from deleting old stuff in `loot-core` as well as some updates to `eslint.js`

**Comment by MatissJanis** [comment]
👋 This looks great btw!

But would you mind splitting it up into smaller PRs? A lot of this stuff we'd be happy to merge NOW, but some other might require more investigations. For example: all the removals of `export` keyword - those are easy to kill off.

**Comment by Shazib** [comment]
> 👋 This looks great btw!
> 
> But would you mind splitting it up into smaller PRs? A lot of this stuff we'd be happy to merge NOW, but some other might require more investigations. For example: all the removals of `export` keyword - those are easy to kill off.

Sure thing, how about one PR for all deleted functions, one for all 'export' removals and one for nuking entre files?

Then i can do something similar separately for loot-core as that is quite a lot in itself.

**Comment by MatissJanis** [comment]
That sounds great! The smaller the PRs and easier to review - the quicker turnaround time you can expect from us (maintainers) :)

**Pull request closed without merging**